src/main/java/co/cask/cdap/guides/traffic/TrafficEventParser.java

```
@@ -0,0 +1,99 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.guides.traffic;
+
+import co.cask.cdap.api.annotation.ProcessInput;
+import co.cask.cdap.api.flow.flowlet.AbstractFlowlet;
+import co.cask.cdap.api.flow.flowlet.OutputEmitter;
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import co.cask.cdap.api.metrics.Metrics;
+import com.google.common.base.Charsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Parses stream events for traffic sensor events in the format:
+ * {@code <segmentID>, <timestamp>, <type>, <count>} (comma-separated values), where each of the fields are:
+ * <ul>
+ *   <li>{@code segmentID} - a unique string identifier for the given road segment the event applies to</li>
+ *   <li>{@code timestamp} - the timestamp for the event in {@code yyyy-MM-dd HH:mm:ss} format</li>
+ *   <li>{@code type} - the type of event (VEHICLE|ACCIDENT)</li>
+ *   <li>{@code count} - the count of incidents to report for the event.</li>
+ * </ul>
+ */
+public class TrafficEventParser extends AbstractFlowlet {
+  /** Special timestamp string used to indicate that the current timestamp should be used */
+  public static final String TIMESTAMP_NOW = "now";
+  /** Expected date format for the timestamp strings */
+  public static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
+
+  /** Metrics key used to count bad records received from the stream */
+  private static final String BAD_EVENT = "event.bad";
```

The metrics use here I think falls under the "good practices".  I think we want to example to mostly focus on timeseries storage, but it's worth still keeping a bit more standards in place in not silently ignoring bad records or errors, etc.

We could simplify some of the error handling here more, but I think there is still some value in illustrating good input checking.